readyapi4j-swagger/src/main/java/com/smartbear/readyapi4j/swagger/SwaggerTestStepBuilder.java

```diff
@@ -91,31 +84,87 @@ private Swagger parseSwagger(String swaggerUrl) throws IllegalArgumentException
     }
 
     /**
+     * Creates a RestRequestStepBuilder for the specified operation with body
+     *
+     * @param operationId the operationId of an operation in the Swagger definition
+     * @return a RestRequestStepWithBodyBuilder with the correct path and method
+     * @throws IllegalArgumentException if the operationId is not found in the Swagger definition
+     */
+    public RestRequestStepWithBodyBuilder operationWithBody(String operationId) {
+        for (Map.Entry<String, Path> path : swagger.getPaths().entrySet()) {
+
+            for(Map.Entry<HttpMethod, Operation> method: path.getValue().getOperationMap().entrySet()) {
+                Operation swaggerOperation = method.getValue();
+                if (!operationId.equalsIgnoreCase(swaggerOperation.getOperationId())) {
+                    continue;
+                }
+                final TestSteps.HttpMethod verb = toHttpMethod(method.getKey());
+                ensureHttpMethodWithBody(verb);
+                ensureBodyParameter(swaggerOperation);
+
+                return new RestRequestStepWithBodyBuilder(
+                        targetEndpoint + swagger.getBasePath() + path.getKey(), verb);
+            }
+        }
+
+        throw new IllegalArgumentException("operationId [" + operationId + "] not found in Swagger definition" );
+    }
+
+    /**
      * Creates a RestRequestStepBuilder for an arbitrary path and method - the targetEndpoint and basePath will be prefixed
      * to the specified path.
      *
      * @param path the path to append to the targetEndpoint and basePath
      * @param method the HTTP method to use
      */
-    public RestRequestStepBuilder<? extends RestRequestStepBuilder> request(String path, TestSteps.HttpMethod method){
+    public RestRequestStepBuilder<RestRequestStepBuilder> request(String path, TestSteps.HttpMethod method){
         String basePath = swagger.getBasePath();
         if( basePath == null ){
             basePath = "";
         } else if( basePath.endsWith("/")){
             basePath = basePath.substring(0, basePath.length()-1);
         }
 
-        if( isHttpMethodWithBody( method )){
-            return new RestRequestStepWithBodyBuilder(targetEndpoint + basePath + path, method);
+        return new RestRequestStepBuilder<>(targetEndpoint + basePath + path, method);
+    }
+
+    /**
+     * Creates a RestRequestStepWithBodyBuilder for an arbitrary path and method - the
+     * targetEndpoint and basePath will be prefixed to the specified path.
+     *
+     * @param path   the path to append to the targetEndpoint and basePath
+     * @param method the HTTP method to use
+     */
+    public RestRequestStepWithBodyBuilder requestWithBody(String path, TestSteps.HttpMethod method){
+        String basePath = swagger.getBasePath();
```

Comment:
suggest we extract this logic to get the basePath into a separate method and call it from here and above